shell/components/SortableTable/index.vue

```diff
@@ -298,6 +298,26 @@ export default {
       type:    String,
       default: 'sortable-table'
     },
+
+    setPageFn: {
```

Comment:
You're right of course, it'd be far simpler if we broke out the filtering/pagination controls from the table itself and instead left it as a simple component that simply rendered rows passed in from above. That being said, I felt that refactoring such a large component was outside of the scope of this particular PR and instead went with the next-best option of passing down a closure that would set the parameters where I need them to kick off a request with those parameters rather than chain up a bunch of events from through multiple components. This approach has the added benefit of making it slightly easier for us to pull code out of these child components at a later date since we can pass in functions that are useful to the specific use-case from the parent component that are specific to how they're being used.